### ISSUE TITLE
Special Players : Do commands and other basic helpers

### ISSUE THREAD

**Issue by k3nnyq3** [The-Purgery/The-Purgery#30]
To help with day-to-day tasks, special players could use some streamlining by expanding their access to shortcut commands for extisting powers such as:

1. Switch : 2 players need to switch in order to balance teams, they however wont.  Current options special players have are !red ID, !blue ID,!swap ID ID. Issue here is that this requires bringin up the playerlist, and if server is full, it takes a minute looking up the needed players and it will likely cause you to sit in the console for the round and also possibility to swap wrong players / mid round etc. 
solution: enable !do_now and !do_later for switch callvotes. Quick, easy and reliable. 

2. Shuffle before the game : This one is not a very big deal as it usually gets f1'd quickly enough anyways. If special players could pass or veote such votes ( in cases multiple shuffles are called with no new players present). Currently special players can shuffle using 2 commands in the correct order : !shuffle (randomizes the players betweeen teams without elo consideration) and after that, !balance (does the obvious). I suppose another solution would be changing the command !shuffle in a way that it'd do what /cv shuffle does. 

3. Teamsize (!ts) - Right now, as the voting system freezes itself and nobody can call votes, it'd help out a lot if special players could stand in and help temporarily change team sizes. There's a way right now of manually moving queued players into teams, using !red and !blue commands, but that is also slow cause of the ID lookup. 

So basically, no "new" permissions, just more refined and streamlined existing solutions to be made available.

K3nny

**Comment by fenix849** [comment]
I agree with k3nny regarding access to !do_later to force swaps when it's over 50 and players are being obstinate about it.

I dont really see an issue with having to !shuffle and !balance but eh.

I think !ts could be handy but I feel like just fixing the vote lockout bug is more important.

**Comment by tjone270** [comment]
Hi @k3nnyq3 and @fenix849, 

1. I've changed `!do_later` to be a Special Player command (previously restricted to Referee level and above.)
2. I think these are seperate functions. The only reason `/cv shuffle` does perform a balance afterwards is this is likely the desired outcome for non-expert players. For us moderators, I personally have F4 bound like this `/bind F4 "!shuffle; wait 40; !balance"` (not totally sure the `wait 40` is actually necessary) which performs the same actions as a `/cv shuffle`. I note that you would not be able to bind this as I can with the way client command handling works, so I've modified it to accommodate these kinds of binds. You should be able to bind this now and use it as I do.
3. The issue causing the voting system to lock up should be resolved, so this should be sorted already.

These changes will take effect upon the next server restart.
